plugins/modules/nmcli.py

```diff
@@ -2604,7 +2640,14 @@ def main():
                               parent=dict(type='str', required=True),
                               promiscuous=dict(type='bool'),
                               tap=dict(type='bool'))),
-            wireguard=dict(type='dict'),
+            wireguard=dict(type='dict',
+                           suboptions=dict(
+                               wgpeer_public_key=dict(type='str', required=False),
+                               wgpeer_allowed_ips=dict(type='str', required=False),
+                               wgpeer_endpoint=dict(type='str', required=False),
+                               wgpeer_persistent_keepalive=dict(type='int', required=False)
```

Comment:
Considering that `wireguard` has no suboptions right now, I'm not sure why you actually need to add support for these new options, instead of simply documenting them with their right name? You should already be able to use them with the current version of the module.